homeassistant/helpers/script.py

```diff
@@ -1123,29 +1141,17 @@ async def async_run(self) -> None:
         """Run script."""
         # Wait for previous run, if any, to finish by attempting to acquire the script's
         # shared lock. At the same time monitor if we've been told to stop.
-        lock_task = self._hass.async_create_task(
-            self._script._queue_lck.acquire()  # pylint: disable=protected-access
-        )
-        stop_task = self._hass.async_create_task(self._stop.wait())
         try:
-            await asyncio.wait(
-                {lock_task, stop_task}, return_when=asyncio.FIRST_COMPLETED
-            )
-        except asyncio.CancelledError:
+            async with async_interrupt.interrupt(self._stop, ScriptStoppedError, None):
```

Comment:
This entire function body except the setting the flag and super call is replaced with `async_interrupt.interrupt` inside a try/catch since they do the same thing